plugins/wasi_nn/wasinnfunc.cpp

```diff
@@ -571,6 +640,50 @@ Expect<uint32_t> WasiNNSetInput::body(const Runtime::CallingFrame &Frame,
 #else
     spdlog::error("[WASI-NN] PyTorch backend is not built. use "
                   "-WASMEDGE_PLUGIN_WASI_NN_BACKEND=\"PyTorch\" to build it.");
+#endif
+  } else if (CxtRef.GraphRef.GraphBackend == WASINN::Backend::TensorflowLite) {
+#ifdef WASMEDGE_PLUGIN_WASI_NN_BACKEND_TFLITE
+    uint32_t InCnt = TfLiteInterpreterGetInputTensorCount(CxtRef.TFLiteInterp);
+    if (Index >= InCnt) {
+      spdlog::error("[WASI-NN] Invalid index id {} for the input, only {} "
+                    "inputs are allowed",
+                    Index, InCnt);
+      return static_cast<uint32_t>(WASINN::ErrNo::InvalidArgument);
+    }
+    uint32_t *Tensor = MemInst->getPointer<uint32_t *>(TensorPtr, 5);
```

Comment:
It seems like the `Tensor[2]` for the tensor type is not used?
Maybe we should give an additional check with the tflite API with the tensor type?